### PR TITLE
feat(pie-monorepo): modified CI Workflow for cater for staging environment

### PR DESCRIPTION
**Pull request by siggerzz**
## Describe your changes (can list changeset entries if preferable)
[Added] - Modified CI Workflow for cater for staging environment

## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: .github/workflows/amplify-deploy.yml [hunk 1]
```diff
@@ -29,16 +29,20 @@ on:
...
+        type: string
 
 env:
   AMPLIFY_ID: ${{ inputs.amplify-app-id }}
   AWS_ACCESS_KEY_ID: ${{ secrets.AWS_KEY_ID }}
   AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
   AWS_REGION: 'eu-west-1'
-  BRANCH_NAME: ${{ github.event_name == 'pull_request' && format('pr{0}', github.event.number) || (github.ref == 'refs/heads/main' && 'main' || github.ref == 'refs/heads/master' && 'master') }}
-  BUCKET_NAME: ${{ github.event_name == 'pull_request' && inputs.bucket-name-preview || (github.ref == 'refs/heads/main' && inputs.bucket-name-main || github.ref == 'refs/heads/master' && inputs.bucket-name-main) }}
+  BRANCH_NAME: ${{ github.event_name == 'pull_request' && format('pr{0}', github.event.number) || (github.ref == 'refs/heads/main' && 'main' || github.ref == 'refs/heads/master' && 'master') || (github.ref == 'refs/heads/staging' && 'staging') }}
```
Comment (jamieomaguire) [inline .github/workflows/amplify-deploy.yml:42]: do we need to reference master here?

**Review by jamieomaguire — DISMISSED** [review]

**Review by JoshuaNg2332 — APPROVED** [review]

**Review by kevinrodrigues — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -122,26 +122,33 @@ If your change is intended to be released under the `latest` tag on npm, you mus
...
-- Create another branch, off this initial feature/beta branch, to implement your code changes. Ensure that this branch **does not** use a prefix.
-- When you create your PR, target the `feature-*` / `beta-*` branch.
-- Upon merging to your `feature-*` / `beta-*` branch, a new PR titled **release: release Packages (beta)** / **release: release Packages (next)** is automatically created. This PR includes the `CHANGELOG.md` and `package.json` version bump. Merging this PR will execute a publish to npm using the appropriate tag.
 
-**Notes:** Any new PRs that target the `feature-*` / `beta-*` branch will cause GitHub actions to include the changes as part of that beta/feature release.
-Any package that uses the `beta` / `next` tag **must** follow this workflow until it's ready to be promoted to the `latest` tag (see Stable Versions section). PRs that combine changes in `latest` and `beta` / `next` packages will result in the beta / next package being versioned incorrectly.
+**Notes:** Any new PRs that target the `beta-*` / `staging` branch will cause GitHub actions to include the changes as part of that beta release.
```
Comment (webbertakken) [inline README.md:143]: A small suggestion.

What I usually do is refer to a version number as a "release" and anything that you deploy as a "deployment".

Releases always have deployments. Deployments are not always for a release as they may not always have a version number attached to it (for example when you deploy a PR's feature branch).

```suggestion
**Notes:** Any new PRs that target the `beta-*` / `staging` branch will cause GitHub actions to include the changes as part of that beta deployment.
```

**Review by webbertakken — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (siggerzz) [reply README.md:143]: Thanks for this! I just realised I need to change this to `beta / next deployment` :D

Comment (webbertakken) [reply README.md:143]: Hehe I was a bit confused, but I didn't wanna comment while not fully in context of your development lifecycle / CI yet.

File: README.md [hunk 3]
```diff
@@ -122,26 +122,33 @@ If your change is intended to be released under the `latest` tag on npm, you mus
 
 A Beta release is a release that contains experimental changes. These are ready for early adoption and testing by consumers but may introduce bugs (or be considered work-in-progress).
 
+If your change is intended to be released under the `beta` tag on npm, you must follow this workflow:
+- Create a new branch with the `beta-*` prefix, and push this to the remote. E.g. `git push origin feature-myawesomework`.
```
Comment (ashleynolan) [inline README.md:126]:
```suggestion
- Create a new branch with the `beta-*` prefix, and push this to the remote. E.g. `git push origin beta-myawesomework`.
```

**Comment by siggerzz** [comment]
Closing this temporarily as work has been paused on using this workflow in fozzie-components.

**Pull request closed without merging**